### PR TITLE
internal/apps: move setup-smoke-test

### PR DESCRIPTION
**Pull request by felixge**
Move setup-smoke-test from internal/apps/setup-smoke-test to internal/setup-smoke tests. This allows the other internal/apps to use the latest version of go. We want this because these apps are often used to test the latest new features which may depend on new go version. They are also used for screenshots where we don't want to show an old go version.

setup-smoke-test is serving a different purpose and needs to be tested against the minimum version of Go that we support. So move it to a separate location and give it its own go.mod file.

### Reviewer's Checklist

- [ ] Changed code has unit tests for its functionality at or near 100% coverage.
- [ ] [System-Tests](https://github.com/DataDog/system-tests/) covering this feature have been added and enabled with the va.b.c-dev version tag.
- [ ] There is a benchmark for any new code, or changes to existing code.
- [ ] If this interacts with the agent in a new way, a system test has been added.
- [ ] Add an appropriate team label so this PR gets put in the right place for the release notes.
- [ ] Non-trivial go.mod changes, e.g. adding new modules, are reviewed by @DataDog/dd-trace-go-guild.

Unsure? Have a question? Request a review!

### CODE REVIEW

File: internal/setup-smoke-test/go.mod [hunk 1]
```diff
@@ -0,0 +1,42 @@
+module github.com/DataDog/dd-trace-go/internal/setup-smoke-test
+
+go 1.22
```
Comment (RomainMuller) [inline internal/setup-smoke-test/go.mod:3]:
```suggestion
go 1.22.0
```

Comment (felixge) [reply internal/setup-smoke-test/go.mod:3]: See https://go.dev/play/p/KkX12rqpSwI - I'll update this PR.

Comment (felixge) [reply internal/setup-smoke-test/go.mod]: 83b630a5e

**Review by RomainMuller — APPROVED** [review]
